### PR TITLE
added a loading indicator to the apply_job page

### PR DESCRIPTION
**Pull request by majakudlicka**
If it's ok I'll copy the logic to other pages with similar functionality

### CODE REVIEW

File: react-ui/src/utils/loading_indicator.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
+//A loading indicator --> from https://github.com/expo/react-loading-indicator
```
Comment (des-des) [inline react-ui/src/utils/loading_indicator.js]: Why have you done this? Why not just import the module?

Comment (majakudlicka) [inline react-ui/src/utils/loading_indicator.js]: Oh I didn't realise you can do that. Updated

File: react-ui/src/reducers/apply_job.js [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
-        ...state,
-        status: action.status,
-        error: action.error,
-        applied: [...state.applied, action.response]
-      };
+        if (action.status === 'pending') {
+            return {
+                ...state,
+                status: action.status,
+                isLoading: true
```
Comment (des-des) [inline react-ui/src/reducers/apply_job.js]: No. 

Use `state.applyJob.status === 'pending'` inside your view, that is exactly what it is for

File: react-ui/src/components/pages/JobDetail.js [hunk 3]
```diff
@@ -96,6 +102,7 @@ function mapStateToProps(state, ownProps) {
     return {
         applied: state.applyJob.applied,
         job_id: ownProps.params.id,
+        isLoading: state.applyJob.isLoading,
```
Comment (des-des) [inline react-ui/src/components/pages/JobDetail.js]: `state.applyJob.status === 'pending'`

**Pull request merged**